src/app/dataset-view/dataset.component.ts

```diff
@@ -55,6 +59,16 @@ export class DatasetComponent extends BaseDatasetDataComponent implements OnInit
             });
 
         this.datasetPermissions$ = this.datasetSubsService.permissionsChanges;
+        this.fetchHashHead(urlDatasetInfo);
+    }
+
+    private fetchHashHead(datasetInfo: DatasetInfo): void {
+        this.datasetService
+            .requestDatasetHeadBlockHash(datasetInfo.accountName, datasetInfo.datasetName)
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((blockHash: string) => {
+                this.currentHeadBlockHash = blockHash;
+            });
```

Comment:
This does not look like presentation level logic to me, should be in some service.

Reply:
Moved to service